src/wallet/wallet2.h

```diff
@@ -1093,6 +1093,9 @@ namespace tools
     void process_new_transaction(const crypto::hash &txid, const cryptonote::transaction& tx, const std::vector<uint64_t> &o_indices, uint64_t height, uint64_t ts, bool miner_tx, bool pool, bool double_spend_seen);
     void process_new_blockchain_entry(const cryptonote::block& b, const cryptonote::block_complete_entry& bche, const crypto::hash& bl_id, uint64_t height, const cryptonote::COMMAND_RPC_GET_BLOCKS_FAST::block_output_indices &o_indices);
     void detach_blockchain(uint64_t height);
+    
+    void detach_blockchain_unsafe(uint64_t height);
```

Comment:
Why the need for a new function? Plus, why all the added whitespace?

Reply:
- when detach_blockchain is called while replaying new blocks, checkpoints sanity check is required. 
- when detach_blockchain is called from import_multisig_info checkpoints sanity check is superfluous, because some multisig transactions could be before checkpoint. So unsafe version is called

Reply:
But `detach_blockchain` is now just calling `detach_blockchain_unssafe`, which is just a slightly modified version of the previous `detach_blockchain`. Why not just make the change to the original function thus negating the need for an extra call?

Reply:
That could be converted in the bool flag, but I decided that additional flags is more ugly than `_unsafe` version. 

I assume we cannot just remove `THROW_WALLET_EXCEPTION_IF` because we're required to verify checkpoints when downloading new blocks. 

But in `import_multisig_info` case this condition could be violated and so I try to avoid it specially for this case only.

Reply:
Thing is, you should still would want a check, hence why it would be better to edit the exception logic IMO. At the end of the day, you don't want to detach at a bad height.